### PR TITLE
Add ExternalCloudProvider TechPreviewNoUpgrade featureGate

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Implementing feature gate selection to run CCM in openshift. https://github.com/openshift/enhancements/pull/463/files#diff-3e0e2c48e70215076dfe36c13768a823ab7080d929d80292f37db2ef5a2121e8R199

### CODE REVIEW

**Review by michaelgugino — CHANGES_REQUESTED** [review]
We don't need this change.

We should deduce in the KCM operator which options to use based on the platform we're using.  We're not making this configurable for users.

**Comment by Danil-Grigorev** [comment]
> We don't need this change.
> 
> We should deduce in the KCM operator which options to use based on the platform we're using. We're not making this configurable for users.

Would an annotation on this resource be a sufficient/ok temporal change not exposed in the API to trigger the transition? This would be a simple change

File: config/v1/types_feature.go [hunk 1]
```diff
@@ -105,7 +105,9 @@ var FeatureSets = map[FeatureSet]*FeatureGateEnabledDisabled{
 		Enabled:  []string{},
 		Disabled: []string{},
 	},
-	TechPreviewNoUpgrade: newDefaultFeatures().toFeatures(),
+	TechPreviewNoUpgrade: newDefaultFeatures().with(
+		"ExternalCloudProvider",
```
Comment (elmiko) [inline config/v1/types_feature.go:109]: should we have a comment here?
```suggestion
		"ExternalCloudProvider", // sig-cloud-provider, danil-grigorev, elmiko
```

**Review by elmiko — COMMENTED** [review]
i was reading through this code again, and had a question

**Pull request closed without merging**